composer/utils/file_helpers.py

```diff
@@ -99,33 +99,20 @@ def ensure_folder_has_no_conflicting_files(folder_name: Union[str, pathlib.Path]
 
     for file in os.listdir(folder_name):
         match = template.match(file)
-        # Encountered an invalid match
+
         if match is not None:
-            valid_match = True
-            # Check each base unit of time and flag later checkpoints
-            if captured_names['token'] and Time.from_token(int(match.group('token'))) > timestamp.token:
-                valid_match = False
-            elif captured_names['sample'] and Time.from_sample(int(match.group('sample'))) > timestamp.sample:
-                valid_match = False
-            elif captured_names['batch'] and Time.from_batch(int(match.group('batch'))) > timestamp.batch:
-                valid_match = False
-            elif captured_names['epoch'] and Time.from_epoch(int(match.group('epoch'))) > timestamp.epoch:
-                valid_match = False
-            # If epoch count is same, check batch_in_epoch, sample_in_epoch, token_in_epoch
-            elif captured_names['epoch'] and Time.from_epoch(int(match.group('epoch'))) == timestamp.epoch:
-                if captured_names['token_in_epoch'] and Time.from_token(int(
-                        match.group('token_in_epoch'))) > timestamp.token_in_epoch:
-                    valid_match = False
-                elif captured_names['sample_in_epoch'] and Time.from_sample(int(
-                        match.group('sample_in_epoch'))) > timestamp.sample_in_epoch:
-                    valid_match = False
-                elif captured_names['batch_in_epoch'] and Time.from_batch(int(
-                        match.group('batch_in_epoch'))) > timestamp.batch_in_epoch:
-                    valid_match = False
-            if not valid_match:
-                raise FileExistsError(
-                    f'{os.path.join(folder_name, file)} exists and conflicts in namespace with a future checkpoint of the current run.'
-                )
+            match = match.groupdict()
+            for unit, value in match.items():
+                if unit.endswith('_in_epoch'):
+                    if 'epoch' not in match:
+                        raise ValueError(f'{filename} has {{unit}} but not {{epoch}}. Add {{epoch}} for uniqueness.')
+                    if int(match['epoch']) != timestamp.epoch:
+                        continue  # only check _in_epoch if both files have same epoch count
```

Comment:
Why is this necessary?

Reply:
If the epochs don't match, `continue` will skip the check of the `_in_epoch` property in the following lines. Let me know if the flow is a bit unclear.